### PR TITLE
mgr/cephadm: set Ganesha pool/namespace settings in the Dashboard

### PR DESCRIPTION
**Pull request by bk201**
- Config Ganesha pool and namespace settings in Dashboard.
- New `config_dashboard` option in cephadm. If the flag is Flase,
  cephadm won't manage Dashboard settings for daemons. This might be
  useful when users want to config API URLs by themselves or for debugging.

Fixes: https://tracker.ceph.com/issues/45155
Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/services/nfs.py [hunk 1]
```diff
@@ -70,6 +71,29 @@ def create(self, daemon_id, host, spec):
             daemon_id, host, spec))
         return self.mgr._create_daemon('nfs', daemon_id, host)
 
+    def config_dashboard(self, daemon_descrs: List[DaemonDescription]):
+        
+        def __get_set_cmd_dicts(out: str) -> List[dict]:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/services/nfs.py:76]:
```suggestion
        def get_set_cmd_dicts(out: str) -> List[dict]:
```
(strip the d-under?)

Comment (bk201) [reply src/pybind/mgr/cephadm/services/nfs.py]: Fixed. Thanks!

**Comment by bk201** [comment]
If services are destroyed, the settings in Dashboard are not removed.
If there is no daemon, current configuring paths are not entered, need to find other ways to handle this. An issue was created to track this: https://tracker.ceph.com/issues/45982

**Comment by sebastian-philipp** [comment]
green: http://pulpito.ceph.com/swagner-2020-06-11_14:16:54-rados:cephadm-wip-swagner3-testing-2020-06-11-1428-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
> If services are destroyed, the settings in Dashboard are not removed.
> If there is no daemon, current configuring paths are not entered, need to find other ways to handle this. An issue was created to track this: https://tracker.ceph.com/issues/45982

cc @jschmid1

**Comment by DanielQujun** [comment]
@sebastian-philipp  @bk201  hey, Guys, I don't con with this PR fix the issue, If we take a look at the Traceback:
```
AttributeError: 'ServiceDescription' object has no attribute 'service'

```

It certainly says that is a no attribute Exception, I have look through the code on the master branch, the service attribute does not exist in 'ServiceDescription' object, I think there are must be something wrong, could you please inspect it?

**Comment by sebastian-philipp** [comment]
maybe #36683 helps?

**Pull request merged**